### PR TITLE
Enhance media_player.cast documentation with port option

### PR DESCRIPTION
**Pull request by Censored3**
**Description:**
Update media_player.cast documentation to include port: configuration entry and some explanation about using Chromecast Audio groups.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#16161

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/cast.markdown [hunk 1]
```diff
@@ -30,9 +30,20 @@ For example, Cast devices can only be discovered if they are on the same subnet
...
 
 - **host** (*Optional*): Use only if you don't want to scan for devices.
+- **port** (*Optional*): Specify chromecast port number - useful to define Chromecast Audio groups
 - **ignore_cec** (*Optional*) A list of Chromecasts that should ignore CEC data for determining the active input. [See the upstream documentation for more information.](https://github.com/balloob/pychromecast#ignoring-cec-data)
+
+### A note on Chromecast Audio groups
+Chromecast Audio groups are added as long as [the discovery component]({{site_root}}/components/discovery/) is used to discover your devices. If for some reason you wish to define your devices manually (i.e.: when running HA in another network), these groups can be manually added by specifying the group master's IP and the group's port.
+These values can be discovered by running netdisco in your local network, i.e.:
+```
+pip install netdisco
```
Comment (DubhAd) [inline source/_components/cast.markdown:87]: It's unclear whether this needs to be done inside the Python venv, or outside of it. Please update to make it clear. Similarly how would this be done on Hass.io?

**Review by DubhAd — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
@Censored3 Could you please take a look at the RFC by @DubhAd? 👍

**Comment by Censored3** [comment]
Sorry for the delay, guys, been a bit busy.

For a venv install, I can add the documentation. But for Hass.io installs, I'm wondering how to instruct the users to do it. I also run on Hass.io and created a simple netdisco Docker container for the sole purpose of doing a single netdisco run (it's available here: https://hub.docker.com/r/censored/netdisco/builds/).

Any suggestions on how to instruct users to run netdisco with hassio?

File: source/_components/cast.markdown [hunk 2]
```diff
@@ -30,9 +30,54 @@ For example, Cast devices can only be discovered if they are on the same subnet
...
+##### {% linkable_title Native installation %}
+On a system in your network with python and pip installed, the following commands can be run:
+```
+pip install netdisco
+python -m netdisco
+```
+and look for the devices listed under `google_cast:`.
+
+##### {% linkable_title Virtualenv installation %}
+If your Home Assistant installation is running in a [virtualenv](https://www.home-assistant.io/docs/installation/virtualenv/), you'll need to activate the virtualenv before running the above commands, e.g.:
```
Comment (DubhAd) [inline source/_components/cast.markdown:93]: One problem here, sorry, is that it doesn't mention running it as the user you run Home Assistant as. Anybody following the RPi guide, or Hassbian, or using anything other than their default user will break their install here.

Comment (Censored3) [reply source/_components/cast.markdown:93]: @DubhAd 
Good catch. I'm not that familiar with anything non-hassio tbh :-)

Can you recheck please?
Also, any suggestions on what to advice hass.io users...?

**Review by DubhAd — DISMISSED** [review]
👍

File: source/_components/cast.markdown [hunk 3]
```diff
@@ -30,9 +30,58 @@ For example, Cast devices can only be discovered if they are on the same subnet
 cast:
   media_player:
   - host: 192.168.1.10
+  - port: 8009
 ```
 
 Configuration variables:
 
 - **host** (*Optional*): Use only if you don't want to scan for devices.
```
Comment (tjorim) [inline source/_components/cast.markdown]: Host is required according to the code

File: source/_components/cast.markdown [hunk 4]
```diff
@@ -30,9 +30,58 @@ For example, Cast devices can only be discovered if they are on the same subnet
...
+- **port** (*Optional*): Specify chromecast port number - useful to define Chromecast Audio groups
```
Comment (tjorim) [inline source/_components/cast.markdown]: Declare the default port

**Review by tjorim — CHANGES_REQUESTED** [review]
Well written first doc PR, @Censored3!
Please, take a look at my remarks and fix the merge conflict

File: source/_components/cast.markdown (hunk 3 above)
Comment (Censored3) [reply source/_components/cast.markdown]: Host is required *if* manually specifying chromecasts.

Just spotted an error however; `port` may not have a `-`.

File: source/_components/cast.markdown (hunk 4 above)
Comment (Censored3) [reply source/_components/cast.markdown]: Done

File: source/_components/cast.markdown [hunk 5]
```diff
@@ -35,6 +35,7 @@ you want to configure the IP address of the Cast device directly:
 cast:
   media_player:
   - host: 192.168.1.10
+    port: 8009
```
Comment (frenck) [inline source/_components/cast.markdown:38]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/cast.markdown [hunk 6]
```diff
@@ -65,3 +70,50 @@ cast:
   - host: IP_ADDRESS_DEVICE_1
   - host: IP_ADDRESS_DEVICE_2
 ```
+
+### {% linkable_title Chromecast Audio Groups and static configuration %}
```
Comment (frenck) [inline source/_components/cast.markdown:74]:
```suggestion
### {% linkable_title Chromecast Audio Groups and static configuration %}

```

File: source/_components/cast.markdown [hunk 7]
```diff
@@ -65,3 +70,50 @@ cast:
...
+Chromecasts and its Audio Groups are added automatically as long as [the discovery component]({{site_root}}/components/discovery/) is used to discover your Chromecast devices, Home Assistant is running in the same network as your Chromecast devices, and you don't have any Chromecast devices configured manually in your *configuration.yaml* file (the configuration includes only `cast:` and no static devices).
+
+If for you wish to define your devices manually (i.e.: when running Home Assistant in another network), these groups can be manually added by specifying the group master's IP and the group's port.
```
Comment (frenck) [inline source/_components/cast.markdown:77]:
```suggestion
If for you wish to define your devices manually (i.e., when running Home Assistant in another network), these groups can be manually added by specifying the group master's IP and the group's port.
```

File: source/_components/cast.markdown [hunk 8]
```diff
@@ -65,3 +70,50 @@ cast:
...
+
+**Caution:** The group is always "tracked" by a single master device. The master of a group could potentially change, after which a manual update of the configuration is required.
+
+#### {% linkable_title Discovering Chromecasts and Audio Groups %}
+The information to use in a static config could be found by e.g. by running a tool called **netdisco** manually in your local network.
```
Comment (frenck) [inline source/_components/cast.markdown:82]:
```suggestion
The information to use in a static config could be found by, e.g., running a tool called `netdisco` manually in your local network.
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by frenck** [comment]
Closing this PR, since the parent PR has been closed.

**Pull request closed without merging**